lte/gateway/c/oai/tasks/mme_app/mme_app_apn_selection.c

```diff
@@ -31,12 +31,94 @@
 #include "mme_app_apn_selection.h"
 #include "emm_data.h"
 
+//------------------------------------------------------------------------------
+int select_pdn_type(
+    struct apn_configuration_s* apn_config,
+    esm_proc_pdn_type_t ue_selected_pdn_type, esm_cause_t* esm_cause) {
+  /* Overwrite apn_config->pdn_type based on the PDN type sent by UE and the PDN
+   * Type received in subscription profile
+   */
```

Comment:
why do we need to overwrite apn_config?

Reply:
The APN config stored in ue_context is the final selected apn_config which will be used in places e.g to create pdn context and to send create session request. mme_app_get_apn_config() function is used to retrieve this apn_config. Hence I am updating the PDN type here.